File: Binary32.java
import java.util.Scanner;

public class Binary32 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String input1 = "";
        String input2 = "";

        try {
            System.out.println("Enter a binary mantissa and exponent (format: mantissax2^exponent):");
            input1 = scanner.nextLine();

            System.out.println("Enter a decimal and exponent (format: decimalx10^exponent):");
            input2 = scanner.nextLine();

            if (input1.isEmpty() && input2.isEmpty()) {
                throw new IllegalArgumentException("Both inputs are empty. At least one should be filled up.");
            }

            if (!input1.isEmpty()) {
                String[] parts1 = input1.split("x2\\\\^");
                if (parts1.length != 2) {
                    throw new IllegalArgumentException("Invalid format. Please enter in the format: mantissax2^exponent");
                }
                String binaryMantissa = parts1[0].trim();
                if (!binaryMantissa.matches("[01]+")) {
                    throw new IllegalArgumentException("Invalid mantissa. Mantissa should only contain 0s and 1s.");
                }
                int exponent1 = Integer.parseInt(parts1[1].trim());
            }

            if (!input2.isEmpty()) {
                String[] parts2 = input2.split("x10\\\\^");
                if (parts2.length != 2) {
                    throw new IllegalArgumentException("Invalid format. Please enter in the format: decimalx10^exponent");
                }
                float decimal = Float.parseFloat(parts2[0].trim());
                if (Float.isNaN(decimal)) {
                    throw new IllegalArgumentException("Invalid decimal. Decimal cannot be NaN.");
                }
                int exponent2 = Integer.parseInt(parts2[1].trim());
            }
        } catch (NumberFormatException e) {
            System.out.println("Invalid number format. Please make sure you're entering numbers correctly.");
        } catch (IllegalArgumentException e) {
            System.out.println(e.getMessage());
        }
    }
}

File: Binary32.js
import React, { useState, useEffect } from 'react';

const InputComponent = () => {
    const [input, setInput] = useState('');
    const [base, setBase] = useState('2');
    const [exponent, setExponent] = useState('');

    useEffect(() => {
        try {
            if (input === '' || exponent === '') {
                throw new Error('Input and exponent cannot be empty.');
            }

            if (base === '2') {
                if (!/^[01]+$/.test(input)) {
                    throw new Error('Invalid input. Input should only contain 0s and 1s for base 2.');
                }
            } else if (base === '10') {
                const decimal = parseFloat(input);
                if (isNaN(decimal)) {
                    throw new Error('Invalid input. Input should be a valid decimal for base 10.');
                }
            }

            const parsedExponent = parseInt(exponent, 10);
            if (isNaN(parsedExponent)) {
                throw new Error('Invalid exponent. Exponent should be a valid integer.');
            }
        } catch (e) {
            console.error(e.message);
        }
    }, [input, base, exponent]);

    return (
        <div>
            <label>
                Input:
                <input type="text" value={input} onChange={e => setInput(e.target.value)} />
            </label>
            <label>
                Base:
                <select value={base} onChange={e => setBase(e.target.value)}>
                    <option value="2">2</option>
                    <option value="10">10</option>
                </select>
            </label>
            <label>
                Exponent:
                <input type="text" value={exponent} onChange={e => setExponent(e.target.value)} />
            </label>
        </div>
    );
};

export default InputComponent;
